backend/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { Controller, Get, Req, Res, UseGuards } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { AuthService } from './auth.service';
+import { GoogleGuard } from './guards/google.guard';
+import { JwtGuard } from './guards/jwt.guard';
+import { JwtRefreshGuard } from './guards/jwtrefresh.guard';
+import { IntraGuard } from './guards/intra.guard';
+
+@Controller()
+export class AuthController {
+  constructor(private AuthService: AuthService) {}
+
+  @Get('google')
+  @UseGuards(GoogleGuard)
+  async googleAuth() {}
+
+  @Get('google/redirect')
+  @UseGuards(GoogleGuard)
+  async googleAuthRedirect(@Req() req: Request, @Res() res: Response) {
+	const { accessToken, refreshToken } = await this.AuthService.login(req.user);
+    res.cookie('jwt', accessToken, {
+      httpOnly: true,
+      sameSite: 'lax',
+    }).cookie('refresh', refreshToken, {
+	  httpOnly: true,
+	  sameSite: 'lax',
+	})
+    return res.send({
+	  accesstoken: accessToken,
+	  refreshtoken: refreshToken
+	});
+  }
+
+  @Get('intra')
+  @UseGuards(IntraGuard)
+  async intraAuth() {}
+
+  @Get('intra/redirect')
+  @UseGuards(IntraGuard)
```

Comment:
If the user sent a GET request to '/intra/redirect?code=blabla' the Intra guard will throw either 'Bad request' or 'Malformed auth code' which you did not catch